File: SDKs/amazonaws.com/iot/2015-05-28/python/src/sdk/models/shared/describecacertificateresponse.py
from dataclasses import dataclass, field
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class DescribeCaCertificateResponse:
    r"""DescribeCaCertificateResponse
    The output from the DescribeCACertificate operation.
    """
    
    certificate_description: Optional[CaCertificateDescription] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('certificateDescription') }})
    registration_config: Optional[RegistrationConfig] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('registrationConfig') }})
    

File: SDKs/amazonaws.com/iot/2015-05-28/typescript/src/sdk/models/shared/describecacertificateresponse.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { CaCertificateDescription } from "./cacertificatedescription";
import { RegistrationConfig } from "./registrationconfig";



// DescribeCaCertificateResponse
/** 
 * The output from the DescribeCACertificate operation.
**/
export class DescribeCaCertificateResponse extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=certificateDescription" })
  certificateDescription?: CaCertificateDescription;

  @SpeakeasyMetadata({ data: "json, name=registrationConfig" })
  registrationConfig?: RegistrationConfig;
}
